### PR TITLE
Port to Gtk+3

### PR DESCRIPTION
**Pull request by yell0wfl4sh**
@quozl  please review!

### CODE REVIEW

**Comment by tony37** [comment]
It seems incredible to see this many changes to the activity, but no 
increment to the version number. This seems to be a consistent pattern - 
a port to gtk+3 in github/SugarLabs with the same version number as the 
original in ASLO. If we were really to take advantage of git, the 
earliest version in ASLO should be committed first. Each subsequent 
version should be submitted as a new version. This would retain the 
development history of the activity. The port to gtk+3 would then be 
submitted as a new version retaining the software history.

Tony

**Comment by yell0wfl4sh** [comment]
@tony37 Thanks for suggestions, I will bump activity version number on all required activities which I have ported. I will also add this on my to-do checklist while porting activities to Gtk+3.

**Comment by chimosky** [comment]
@tony37 I agree with what you're saying but I think it's because he hasn't released any yet, and he'll definitely change version numbers before releasing any activity.

**Comment by quozl** [comment]
We are not using git as a version control system only; we're also using it as a change control system.  So version number changes are only needed when a new release of an activity is to be made.

It is up to the contributor whether to include that in their pull request.  I'll accept pull requests with or without version number changes; but if someone includes a version number change, then I'll expect also;
* full testing of the activity features, (for which others can help),
* updated translations,
* updated NEWS file,
* review of licensing,
* updated README.md file,
* several other interesting things, see the [checklist - maintainer](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#checklist---maintainer),
* the pull request to be "vN", where N is the version number.

**Comment by tony37** [comment]
Great!

Tony

**Comment by tony37** [comment]
Essentially, the repository for an activity in Github/SugarLabs is the 
official current working version of an activity. As James Cameron points 
out, changes to an activity is done in the developers own work 
environment - often on their own computer. When a new version is ready, 
it can be submitted as a pull request. The procedure for releasing a new 
version to ASLO has been under development for a long time and 
apparently is not yet ready. Developers other than Walter Bender have 
not released new versions from Github to ASLO although that original 
mechanism is clearly in place and working. As a consequence none of the 
GTK+3 ports is available to our users.

Tony

**Comment by chimosky** [comment]
I tried to upload a new version of an activity to ASLO - words activity - some time ago and I couldn't; the activity just wouldn't upload, this is due to the fact that in recent times ASLO hasn't been maintained and I still cannot upload it now.

**Comment by tony37** [comment]
Are you really saying that the 100+ activities ported to GTK+3 are 
simply changes and do not represent a new version?

When do you expect it will  be possible to release any of these ports. 
So far the only one I know is the release of turtleblocks v 218 by 
Walter Bender. I believe the github/repositories for activities need to 
represent working versions and should at least be one  version number 
above those in ASLO preceding the changes. I see no reason for using 
pull requests for less than a new version.

Let the users test the activity. The activity should be tested to see 
that Browse downloads, installs and that the activity can start in 
Ubuntu Sugar (18.04 LTS) and in build 13.2.9. If the new version does 
not work in one of these environments, add this fact to the release notes.

Tony

**Comment by quozl** [comment]
I'll continue to welcome pull requests that don't change the version number, so that contributors can work on small fixes without having to take on the role of activity maintainer.  These small fixes do help.

However, one of the things Tony said could be paraphrased; unless people make releases, these small fixes won't get used, and contributors work is wasted.

Tony, I disagree that the head of the master branch is to be the currently working released version.  We've not operated our git repositories in that way.  The currently working released version will be a recent git tag, shown by the releases tab.  This is the case for most activity repositories.  This one is different; you created it from bundles without adding release tags.

**Comment by quozl** [comment]
@yashagrawal3, thanks for the pull request, I've reviewed to 66ed873.  Did you test?  I've not had time yet.

**Comment by chimosky** [comment]
@yashagrawal3 reviewd up to 66ed873, activity opens and when `Guide` is clicked the guide is open but there's no way i can go back to the main menu - except i press `esc` and not every user knows that, same goes for other buttons - and after `Guide` is clicked if you try to go to the activity icon on the top left or `stop button` on the top right there's a delay before the activity notices the mouse hover.

**Comment by yell0wfl4sh** [comment]
@chimosky there is `skip` button at the bottom which does the same work, improvements can be done but I  don't think they are related to this pull request, please open new issue you think need a work on.

@quozl  Yes, I have tested the activity properly on Ubuntu 17.10, activity works fine though some errors are logged. I will be following the method that is followed in other activities, that first gets the pull request and then open a pull request to release the activity.
So please review this pull request, I will be releasing later this week or next week.

File: olpcgames/eventwrap.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
         release a few more resources, then a bit more...
 """
 import pygame
-import gtk
+from gi.repository import Gtk
```
Comment (quozl) [inline olpcgames/eventwrap.py:22]: Is the import needed?  I see no reference to it in the file.

File: olpcgames/video.py [hunk 2]
```diff
@@ -14,10 +14,13 @@
 import olpcgames
 from olpcgames import _gtkmain
 
-import pygtk
-pygtk.require('2.0')
-import gtk
-import gst
+import gi
+gi.require_version('Gtk', '3.0')
```
Comment (quozl) [inline olpcgames/video.py:18]: This seems to be a duplicate.  Please put all the necessary require_version calls in the main program?

File: olpcgames/eventwrap.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
...
 import Queue
 import thread, threading
```
Comment (quozl) [inline olpcgames/eventwrap.py:24]: I see an import of Gtk and of threading.  That GTK+ 3 does not support threading very well is one of the main reasons we ported from olpcgames to sugargame.  Don't forget to test this on a multiprocessor system under load.  Consider completing the port to sugargame.

**Review by quozl — COMMENTED** [review]
Thanks.  I looked at the changes, and have comments.

**Pull request closed without merging**